### PR TITLE
Add support for distributing releases to multiple distribution groups

### LINKED ISSUE #662: Allow "appcenter distribute release -g" to send to multiple group
Hello,
is it possible to consider to add this option to upload to multiple groups with the `appcenter distribute release` command ?
Or maybe there is a possibility with the -g option but I didn't find it.

Currently it's possible to do this with the open API by using : 
`curl -X PATCH "https://api.appcenter.ms/v0.1/apps/${OWNERNAME}/${APPNAME}/releases/${LASTEST_RELEASEID}" -H "accept: application/json" -H "X-API-Token:${APPCENTER_TOKEN}" -H "Content-Type: application/json" -d '{ "release_notes": "${RELEASENOTES}", "mandatory_update": false, $DISTRIBUTIONS_JSON, "notify_testers": false}'`

 so I think it could be achieved with appcenter-cli version ?
I mean nobody want to patch the app after submitting (distribute release) an application with a group option that will managed only one group.

And currently it's impossible to update the application groups with the  `appcenter apps update` command anyway :)

I just see a ticket has been closed month ago with the same issue #467 but for me it's not really convenient to switch from one method to another (CLI / API) my main concern is to do only one method.

My explanations :

Because last week I implement really quickly CLI version of our old hockeyapp pipeline. Then I saw it wasn't possible to not notify people.

So I switched to API system, today it was almost over when I realized it was really not convenient to upload a dSym.zip file cause ours are above 256MB and the silent option for release was out last night.

So I switched back to CLI system, integrate the --silent during distribute release process. But then I figure out it was not possible to upload it to multiple group and moreover it's impossible to update it after with the apps update method.

HockeyApp will be dropped within a month (roughly) and the switch is not currently smooth as it must be (according to me). In fact, I don't care if I need to spend a week to switch from one system to another as soon as I don't have to implement X system (CLI AND API) to do what was simple to do with HockeyApp previously.

### PR DESCRIPTION
**Pull request by charliealbright**
This PR adds support for specifying multiple distribution groups when executing the `appcenter distribute release` command. the `-g` command now supports comma-delimited lists of groups. For example:

- `-g thisgroup,thatgroup`
- `-g thisgroup,"group with spaces"`

**Changelog description:**
> Flexible distribution: The CLI tool now supports distributing a release to multiple distribution groups.

I believe I have covered all my bases, but let me know if any changes/additions should be made.

Fixes #662

### CODE REVIEW

**Comment by russelarms** [comment]
Hi @charliealbright! We really appreciate the contribution to this long-living feature request.
Would you mind adding tests for the change to `test/commands/distribute/release/release-test.ts`?

**Comment by charliealbright** [comment]
@russelarms I added tests there and made a new file for testing the utils I wrote for this feature.

Had some trouble with the mocking setup but eventually got it working.

**Review by Jamminroot — DISMISSED** [review]
LGTM

File: src/commands/distribute/release.ts [hunk 1]
```diff
@@ -280,7 +293,8 @@ export default class ReleaseBinaryCommand extends AppCommand {
     let distributionGroupUsersNumber: Promise<number | null>;
     let storeInformation: Promise<models.ExternalStoreResponse | null>;
     if (!_.isNil(this.distributionGroup)) {
-      // get number of distribution group users (and check distribution group existence)
+      debug(`Distro group: ${this.distributionGroup}`);
```
Comment (DmitriyKirakosyan) [inline src/commands/distribute/release.ts:296]: Do we need this log?

Comment (charliealbright) [reply src/commands/distribute/release.ts:296]: No, it's not needed. I can remove it.

**Comment by charliealbright** [comment]
@Jamminroot @DmitriyKirakosyan this should be ready to review again

File: src/commands/distribute/lib/distribute-util.ts [hunk 2]
```diff
@@ -96,3 +96,19 @@ export async function addGroupToRelease(options: AddGroupToReleaseOptions): Prom
     throw failure(ErrorCodes.Exception, `Could not add ${destinationType} ${destination} to release ${releaseId}`);
   }
 }
+
+export function parseDistributionGroups(groups: string): string[] {
+  debug(`Parsing group string: ${groups}`);
```
Comment (DmitriyKirakosyan) [inline src/commands/distribute/lib/distribute-util.ts:101]:
```suggestion
```

Comment (DmitriyKirakosyan) [reply src/commands/distribute/lib/distribute-util.ts:101]: Let's remove this one as well. I guess it was added for debug purpose only.

Comment (charliealbright) [reply src/commands/distribute/lib/distribute-util.ts]: done!

**Review by DmitriyKirakosyan — APPROVED** [review]
LGTM 👍

**Review by russelarms — APPROVED** [review]
Tested it, works well. Thank you so much for the effort you put into this contribution.
We'll merge it once the CI is back to normal (another team is working on this at the moment).

**Comment by szczepanski-k** [comment]
Could you please let me know in which version will the fix be released?

**Comment by charliealbright** [comment]
@russelarms is there a timeline for when the next release of `appcenter-cli` is expected to happen? Looking to use this soon!

**Comment by MatkovIvan** [comment]
Hi @szczepanski-k and @charliealbright,
There is no specific ETA. We're planning to batch a few more fixes to the release.

**Pull request merged**